### PR TITLE
feat(core): add 'relation to type material' as as filtering option

### PR DESCRIPTION
**Pull request by jdaviscooke**
We have a use case where we want to be able to filter bacterial/fungal assemblies based on the values in the 'relation to type material' metadata. This adds that ability while keeping the default behavior unchanged.

### CODE REVIEW

File: ncbi_genome_download/core.py [hunk 1]
```diff
@@ -192,7 +199,7 @@ def config_download(config):
                 return 1
 
         if config.metadata_table:
-            with codecs.open(config.metadata_table, mode='w', encoding='utf-8') as handle:
+            with open(config.metadata_table, 'wt') as handle:
```
Comment (kblin) [inline ncbi_genome_download/core.py]: Why is this needed? With this change you're basically rolling back 8c844ab0 and re-opening issue #78 for Python 2 installs.

File: tests/test_arguments.py [hunk 2]
```diff
@@ -54,3 +54,33 @@ def test_assembly_accessions():
...
+  parser = argument_parser()
+  ns = parser.parse_args(args)
+  assert ns.type_material == 'reference'
+
+  args = ['-M', 'all', 'fungi']
+  parser = argument_parser()
+  ns = parser.parse_args(args)
+  assert ns.type_material == 'all'
+
+  #'type', 'reference', 'synonym', 'proxytype', 'neotype'
```
Comment (kblin) [inline tests/test_arguments.py]: Not sure what this line is doing here.

File: ncbi_genome_download/core.py (hunk 1 above)
Comment (jdaviscooke) [inline ncbi_genome_download/core.py]:
```suggestion
            with codecs.open(config.metadata_table, mode='w', encoding='utf-8') as handle:
```

File: tests/test_arguments.py (hunk 2 above)
Comment (jdaviscooke) [inline tests/test_arguments.py]:
```suggestion
```

File: ncbi_genome_download/core.py (hunk 1 above)
Comment (jdaviscooke) [reply ncbi_genome_download/core.py]: Sorry, looks like that commit came after I forked and I didn't pull/rebase to the latest before submitting the PR. Fixed.

File: tests/test_arguments.py (hunk 2 above)
Comment (jdaviscooke) [reply tests/test_arguments.py]: Stray comment removed

**Comment by kblin** [comment]
Thanks for your contribution!
